### PR TITLE
Fix issue #13728. Create separate connectionManager for different filter chains 

### PR DESCRIPTION
**Pull request by crazyxy**
Issue #13728

Code adding `filter`
https://github.com/istio/istio/blob/7d61672e25a28b0fa49ef2091f5b49703307572f/pilot/pkg/networking/core/v1alpha3/envoyfilter.go#L81

All entries in `httpConnectionManagers` refer to the same object. Only the first `filter_chain` will get the correct number of `http_filter` and others will get more `http_filter` than expected.

### CODE REVIEW

File: pilot/pkg/networking/core/v1alpha3/listener_test.go [hunk 1]
```diff
@@ -419,8 +422,8 @@ func verifyOutboundTCPListenerHostname(t *testing.T, l *xdsapi.Listener, hostnam
 
 func verifyInboundHTTPListenerServerName(t *testing.T, l *xdsapi.Listener) {
 	t.Helper()
-	if len(l.FilterChains) != 1 {
-		t.Fatalf("expected %d filter chains, found %d", 1, len(l.FilterChains))
+	if len(l.FilterChains) != 2 {
```
Comment (howardjohn) [inline pilot/pkg/networking/core/v1alpha3/listener_test.go]: It isn't clear to me why this changed. Why do we have two filter chains now? I thought we have the same filter chain, but use a distinct connectionManager so when we mutate one it doesn't mutate the others. Maybe I misunderstood the change though

Comment (crazyxy) [reply pilot/pkg/networking/core/v1alpha3/listener_test.go]: You are right. We still have the same filter chain and have separate connectionManager. 

Two filter chains are only used for test. I configured the `OnInboundFilterChains` to have two filter chains.

**Comment by crazyxy** [comment]
@costinm gentle ping

**Comment by rshriram** [comment]
This is a good catch. Thanks!

**Pull request merged**